### PR TITLE
Full upgrade

### PR DESCRIPTION
**Pull request by yalh76**
## Solution
- *Apply example_ynh*
- *Update to 1.2.4*

## PR Status
- [x] Code finished.
- [x] Tested with Package_check.
- [x] Fix or enhancement tested.
- [x] Upgrade from last version tested.
- [x] Can be reviewed and tested.

## Validation
---
- [ ] **Code review**
- [ ] **Approval (LGTM)**  
*Code review and approval have to be from a member of @YunoHost/apps group*
- **CI succeeded** : 
[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/phpldapadmin_ynh_PR5%20(yalh76)/)  

When the PR is marked as ready to merge, you have to wait for 3 days before really merging it.

### CODE REVIEW

**Comment by yalh76** [comment]
if someone from @YunoHost-Apps/apps-group can review and merge that PR :)
Thanks in advance

File: LICENSE [hunk 1]
```diff
@@ -1,642 +1,301 @@
-                    GNU GENERAL PUBLIC LICENSE
-                       Version 3, 29 June 2007
+		    GNU GENERAL PUBLIC LICENSE
+		       Version 2, June 1991
```
Comment (maniackcrudelis) [inline LICENSE]: Please do not change the license of the package.
One has been chosen, we should now keep it.

File: pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+- [ ] Tested with Package_check.
+- [ ] Fix or enhancement tested.
+- [ ] Upgrade from last version tested.
+- [ ] Can be reviewed and tested.
+
+## Validation
+---
+- [ ] **Code review**
+- [ ] **Approval (LGTM)**  
+*Code review and approval have to be from a member of @YunoHost/apps group*
```
Comment (maniackcrudelis) [inline pull_request_template.md]: This app isn't a High Quality app, no need to ask to the Apps group for each PR.

File: pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+- **CI succeeded** : 
+[![Build Status](https://ci-apps-dev.yunohost.org/jenkins/job/phpldapadmin_ynh%20-BRANCH-/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/phpldapadmin_ynh%20-BRANCH-/)
```
Comment (maniackcrudelis) [inline pull_request_template.md]: As well, you can't expect any automatic tests for this app.

File: scripts/_common.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+#=================================================
+# COMMON VARIABLES
+#=================================================
+
+# dependencies used by the app
+pkg_dependencies="deb1 deb2"
```
Comment (maniackcrudelis) [inline scripts/_common.sh]: ?

File: scripts/install [hunk 5]
```diff
@@ -1,63 +1,127 @@
...
+
+#=================================================
+# MANAGE SCRIPT FAILURE
+#=================================================
 
-# Source YunoHost helpers
-. /usr/share/yunohost/helpers
+ynh_clean_setup () {
+	read -p "key"
+	### Remove this function if there's nothing to clean before calling the remove script.
```
Comment (maniackcrudelis) [inline scripts/install]: As it's written here, you should remove it.

File: scripts/upgrade [hunk 6]
```diff
@@ -1,59 +1,143 @@
...
+
+#=================================================
+# BACKUP BEFORE UPGRADE THEN ACTIVE TRAP
+#=================================================
+ynh_script_progression --message="Backing up the app before upgrading (may take a while)..." --weight=6
+
+# Backup the current version of the app
+ynh_backup_before_upgrade
+ynh_clean_setup () {
+	read -p "key"
```
Comment (maniackcrudelis) [inline scripts/upgrade]: To be removed

**Comment by maniackcrudelis** [comment]
Small corrections to do.
Otherwise, everything's ok. And it works.

File: LICENSE (hunk 1 above)
Comment (yalh76) [reply LICENSE]: Even if it's not the one in the upstream package file ?
* https://github.com/leenooks/phpLDAPadmin/blob/master/LICENSE

File: scripts/install (hunk 5 above)
Comment (yalh76) [reply scripts/install]: done, forgot to remove my test steps ^^

File: scripts/upgrade (hunk 6 above)
Comment (yalh76) [reply scripts/upgrade]: also done

File: scripts/_common.sh (hunk 4 above)
Comment (yalh76) [reply scripts/_common.sh]: removed ^^, kept it for the day it will be needed ^^

File: LICENSE (hunk 1 above)
Comment (maniackcrudelis) [reply LICENSE]: This license file is for the package itself, not the upstream app.

File: pull_request_template.md (hunk 2 above)
Comment (yalh76) [reply pull_request_template.md]: removed

File: LICENSE (hunk 1 above)
Comment (yalh76) [reply LICENSE]: My bad, I haven't understood the difference between the package license and the upstream application license. I use to use the same for both :/

File: pull_request_template.md (hunk 3 above)
Comment (yalh76) [reply pull_request_template.md]: I agree but ci-apps-dev is not automatic, having the link already setuped allows to easily have the results of a test launched manually on ci-apps-dev. it's a bad idea ?

Comment (maniackcrudelis) [reply pull_request_template.md]: This link won't work as it is. So for me it's not really useful.
The link will without the branch, but will the name of the person who add the job.
Also, unless you're the maintainer, I doubt anyone would use it.

Comment (yalh76) [reply pull_request_template.md]: ok then, as I don't plan to be the maintainer, I removed it ^^

**Pull request merged**